tornado/iostream.py

```diff
@@ -117,6 +117,82 @@ class StreamBufferFullError(Exception):
     """
 
 
+class _StreamBuffer(object):
+
+    def __init__(self):
+        # A sequence of (False, bytearray) and (True, memoryview) objects
+        self._buffers = collections.deque()
+        # Position in the first buffer
+        self._first_pos = 0
+        self._size = 0
+
+    def __len__(self):
+        return self._size
+
+    # Data above this size will be appended separately instead
+    # of extending an existing bytearray
+    _large_buf_threshold = 2048
+
+    def append(self, data):
+        size = len(data)
+        if size > self._large_buf_threshold:
+            if not isinstance(data, memoryview):
+                data = memoryview(data)
+            self._buffers.append((True, data))
+        elif size > 0:
+            if self._buffers:
+                is_large, b = self._buffers[-1]
+                is_large = is_large or len(b) >= self._large_buf_threshold
```

Comment:
`is_large` is doing a lot of different things here. I think it woudl be more clear if this were

```
if self._buffers: 
    is_memview, b = self._buffers[-1]
    new_buf = is_memview or len(b) > self._large_buf_threshold
else:
    new_buf = True
```